asclepias_broker/metadata/api.py

```diff
@@ -50,14 +51,16 @@ def update_metadata_from_event(relationship: Relationship, payload: dict):
 def update_metadata(identifier: str, scheme: str, data: dict,
                     create_identity_events=True,
                     create_missing_groups=True,
-                    provider: str = None, link_publication_date: str = None):
+                    providers: List[str] = None,
+                    link_publication_date: str = None):
     """."""
     from ..events.api import EventAPI
 
     # Check if there are identity links that can be created:
     identifiers = data.get('Identifier', [])
     if create_identity_events and len(identifiers) > 1:
-        provider = provider or 'unknown'
+        providers = [{'Name': provider} for provider in providers] or \
```

Comment:
Since `providers` might be `None`, you should do a `providers = providers or ['unknown']` probably at the beginning